[
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555650879",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-555650879",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 555650879,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NTY1MDg3OQ==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-19T18:44:57Z",
        "updated_at": "2019-11-19T18:44:57Z",
        "author_association": "NONE",
        "body": "Even with the `scipy` installed, more uninstalled dependencies were reflected. How can I build the wheel correctly? Thanks.\r\n```\r\n2019-11-19 18:42:27,098 Build [DEBUG] - Running subprocess in '/tmp/onnxruntime/build/Linux/Debug'\r\n['/usr/bin/python3', '/tmp/onnxruntime/onnxruntime/test/onnx/gen_test_models.py', '--output_dir', 'test_models']\r\n2019-11-19 18:42:31,371 Build [DEBUG] - Running subprocess in '/tmp/onnxruntime/build/Linux/Debug'\r\n['/tmp/onnxruntime/build/Linux/Debug/onnx_test_runner', 'test_models']\r\n2019-11-19 18:42:31.483881866 [E:onnxruntime:Default, runner.cc:171 ParallelRunTests] Running tests in parallel: at most 35 models at any time\r\n2019-11-19 18:42:57.154138575 [E:onnxruntime:Default, runner.cc:190 ParallelRunTests] Running tests finished. Generating report\r\nresult: \r\n\tModels: 35\r\n\tTotal test cases: 35\r\n\t\tSucceeded: 35\r\n\t\tNot implemented: 0\r\n\t\tFailed: 0\r\n\tStats by Operator type:\r\n\t\tNot implemented(0): \r\n\t\tFailed:\r\nFailed Test Cases:\r\n/tmp/onnxruntime/tools/ci_build/build.py:620: UserWarning: onnxmltools and keras are not installed. Following test cannot be run.\r\n  warnings.warn(\"onnxmltools and keras are not installed. Following test cannot be run.\")\r\n2019-11-19 18:42:57,207 Build [DEBUG] - Running subprocess in '/tmp/onnxruntime/build/Linux/Debug'\r\n['/usr/bin/python3', '/tmp/onnxruntime/setup.py', 'bdist_wheel']\r\nTraceback (most recent call last):\r\n  File \"/tmp/onnxruntime/setup.py\", line 220, in <module>\r\n    'Programming Language :: Python :: 3.7'],\r\n  File \"/usr/local/lib/python3.6/dist-packages/setuptools/__init__.py\", line 145, in setup\r\n    return distutils.core.setup(**attrs)\r\n  File \"/usr/lib/python3.6/distutils/core.py\", line 134, in setup\r\n    ok = dist.parse_command_line()\r\n  File \"/usr/local/lib/python3.6/dist-packages/setuptools/dist.py\", line 706, in parse_command_line\r\n    result = _Distribution.parse_command_line(self)\r\n  File \"/usr/lib/python3.6/distutils/dist.py\", line 472, in parse_command_line\r\n    args = self._parse_command_opts(parser, args)\r\n  File \"/usr/local/lib/python3.6/dist-packages/setuptools/dist.py\", line 1021, in _parse_command_opts\r\n    nargs = _Distribution._parse_command_opts(self, parser, args)\r\n  File \"/usr/lib/python3.6/distutils/dist.py\", line 534, in _parse_command_opts\r\n    if not issubclass(cmd_class, Command):\r\nTypeError: issubclass() arg 1 must be a class\r\nTraceback (most recent call last):\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 1043, in <module>\r\n    sys.exit(main())\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 1034, in main\r\n    build_python_wheel(source_dir, build_dir, configs, args.use_cuda, args.use_ngraph, args.use_tensorrt, args.use_openvino, args.use_nuphar, nightly_build)\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 795, in build_python_wheel\r\n    run_subprocess(args, cwd=cwd)\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 196, in run_subprocess\r\n    return subprocess.run(args, cwd=cwd, check=True, stdout=stdout, stderr=stderr, env=my_env, shell=shell)\r\n  File \"/usr/lib/python3.6/subprocess.py\", line 438, in run\r\n    output=stdout, stderr=stderr)\r\nsubprocess.CalledProcessError: Command '['/usr/bin/python3', '/tmp/onnxruntime/setup.py', 'bdist_wheel']' returned non-zero exit status 1.\r\n```",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555650879/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555653234",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-555653234",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 555653234,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NTY1MzIzNA==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-19T18:50:59Z",
        "updated_at": "2019-11-19T18:50:59Z",
        "author_association": "NONE",
        "body": "It seems that I could still install the Python package even though the test was failed. This misleading messages and tests should really be avoided.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555653234/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555653801",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-555653801",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 555653801,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NTY1MzgwMQ==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-19T18:52:26Z",
        "updated_at": "2019-11-19T18:52:43Z",
        "author_association": "NONE",
        "body": "The `--skip_onnx_tests` bug should be fixed and the tests should be done in parallel as well. Some people cannot afford waiting.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555653801/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555692054",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-555692054",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 555692054,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NTY5MjA1NA==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-19T20:14:42Z",
        "updated_at": "2019-11-19T20:14:42Z",
        "author_association": "NONE",
        "body": "Adding `--use_cuda` resulted the following error in the tests after installation.\r\n```\r\n4: [       OK ] OpaqueApiTest.RunModelWithOpaqueInputOutput (66 ms)\r\n4: [----------] 1 test from OpaqueApiTest (66 ms total)\r\n4: \r\n4: [----------] Global test environment tear-down\r\n4: [==========] 1 test from 1 test case ran. (66 ms total)\r\n4: [  PASSED  ] 1 test.\r\n4/4 Test #4: opaque_api_test ..................   Passed    0.08 sec\r\n\r\n75% tests passed, 1 tests failed out of 4\r\n\r\nTotal Test time (real) = 421.45 sec\r\n\r\nThe following tests FAILED:\r\n\t  1 - onnxruntime_test_all (Failed)\r\nErrors while running CTest\r\nTraceback (most recent call last):\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 1043, in <module>\r\n    sys.exit(main())\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 987, in main\r\n    args.use_dnnlibrary)\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 593, in run_onnxruntime_tests\r\n    cwd=cwd, dll_path=dll_path)\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 196, in run_subprocess\r\n    return subprocess.run(args, cwd=cwd, check=True, stdout=stdout, stderr=stderr, env=my_env, shell=shell)\r\n  File \"/usr/lib/python3.6/subprocess.py\", line 438, in run\r\n    output=stdout, stderr=stderr)\r\nsubprocess.CalledProcessError: Command '['/usr/local/bin/ctest', '--build-config', 'Debug', '--verbose']' returned non-zero exit status 8.\r\n```",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555692054/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555706166",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-555706166",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 555706166,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NTcwNjE2Ng==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-19T20:48:08Z",
        "updated_at": "2019-11-19T20:48:08Z",
        "author_association": "NONE",
        "body": "`--config Release` sorts of make the test running in parallel.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555706166/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555783475",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-555783475",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 555783475,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NTc4MzQ3NQ==",
        "user": {
            "login": "skottmckay",
            "id": 979079,
            "node_id": "MDQ6VXNlcjk3OTA3OQ==",
            "avatar_url": "https://avatars.githubusercontent.com/u/979079?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/skottmckay",
            "html_url": "https://github.com/skottmckay",
            "followers_url": "https://api.github.com/users/skottmckay/followers",
            "following_url": "https://api.github.com/users/skottmckay/following{/other_user}",
            "gists_url": "https://api.github.com/users/skottmckay/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/skottmckay/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/skottmckay/subscriptions",
            "organizations_url": "https://api.github.com/users/skottmckay/orgs",
            "repos_url": "https://api.github.com/users/skottmckay/repos",
            "events_url": "https://api.github.com/users/skottmckay/events{/privacy}",
            "received_events_url": "https://api.github.com/users/skottmckay/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-20T00:46:59Z",
        "updated_at": "2019-11-20T00:46:59Z",
        "author_association": "MEMBER",
        "body": "There are 3 phases: update, build and test. Default is to do all three. Running the onnx tests is just a part of all the potential tests.\r\n\r\nAdd '--update --build' to skip all the tests.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555783475/reactions",
            "total_count": 1,
            "+1": 1,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555789487",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-555789487",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 555789487,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NTc4OTQ4Nw==",
        "user": {
            "login": "dashesy",
            "id": 873905,
            "node_id": "MDQ6VXNlcjg3MzkwNQ==",
            "avatar_url": "https://avatars.githubusercontent.com/u/873905?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/dashesy",
            "html_url": "https://github.com/dashesy",
            "followers_url": "https://api.github.com/users/dashesy/followers",
            "following_url": "https://api.github.com/users/dashesy/following{/other_user}",
            "gists_url": "https://api.github.com/users/dashesy/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/dashesy/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/dashesy/subscriptions",
            "organizations_url": "https://api.github.com/users/dashesy/orgs",
            "repos_url": "https://api.github.com/users/dashesy/repos",
            "events_url": "https://api.github.com/users/dashesy/events{/privacy}",
            "received_events_url": "https://api.github.com/users/dashesy/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-20T01:12:02Z",
        "updated_at": "2019-11-20T01:12:02Z",
        "author_association": "NONE",
        "body": "I had the same question in #1938 this means the `--update --build` to skip tests in counter-intuitive when there is a `--skip_onnx_tests` option.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555789487/reactions",
            "total_count": 1,
            "+1": 1,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555861049",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-555861049",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 555861049,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NTg2MTA0OQ==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-20T06:34:17Z",
        "updated_at": "2019-11-20T06:34:17Z",
        "author_association": "NONE",
        "body": "> There are 3 phases: update, build and test. Default is to do all three. Running the onnx tests is just a part of all the potential tests.\r\n> \r\n> Add '--update --build' to skip all the tests.\r\n\r\nThis works. However, I still wish the test bugs could be fixed since the tests are also important for the software.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/555861049/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/556332771",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-556332771",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 556332771,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NjMzMjc3MQ==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-20T20:11:16Z",
        "updated_at": "2019-11-20T20:11:16Z",
        "author_association": "NONE",
        "body": "It is weird. Yesterday I tried to install the ONNX Runtime using the following command in a CUDA-cuDNN container on a RTX-2080TI computer at home, and it worked fine without any error. However, today I used the exact same container on a V100 computer. It produces error.\r\n```\r\ngit clone --recursive https://github.com/Microsoft/onnxruntime\r\ncd onnxruntime\r\ngit checkout rel-1.0.0\r\n./build.sh --config Release --cuda_home /usr/local/cuda --cudnn_home /usr/lib/x86_64-linux-gnu/ --use_cuda --parallel --use_full_protobuf --update --build --build_wheel\r\ncd build/Linux/Release/\r\nmake install\r\npython ../../../setup.py install\r\n```\r\nThe error message is:\r\n```\r\n[100%] Linking CUDA device code CMakeFiles/onnxruntime_pybind11_state.dir/cmake_device_link.o\r\n[100%] Linking CXX shared module onnxruntime_pybind11_state.so\r\n[100%] Built target onnxruntime_pybind11_state\r\n2019-11-20 20:06:24,557 Build [DEBUG] - Running subprocess in '/tmp/onnxruntime/build/Linux/Release'\r\n['/usr/bin/python3', '/tmp/onnxruntime/setup.py', 'bdist_wheel', '--use_cuda']\r\nTraceback (most recent call last):\r\n  File \"/tmp/onnxruntime/setup.py\", line 224, in <module>\r\n    'Programming Language :: Python :: 3.7'],\r\n  File \"/usr/local/lib/python3.6/dist-packages/setuptools/__init__.py\", line 145, in setup\r\n    return distutils.core.setup(**attrs)\r\n  File \"/usr/lib/python3.6/distutils/core.py\", line 134, in setup\r\n    ok = dist.parse_command_line()\r\n  File \"/usr/local/lib/python3.6/dist-packages/setuptools/dist.py\", line 706, in parse_command_line\r\n    result = _Distribution.parse_command_line(self)\r\n  File \"/usr/lib/python3.6/distutils/dist.py\", line 472, in parse_command_line\r\n    args = self._parse_command_opts(parser, args)\r\n  File \"/usr/local/lib/python3.6/dist-packages/setuptools/dist.py\", line 1021, in _parse_command_opts\r\n    nargs = _Distribution._parse_command_opts(self, parser, args)\r\n  File \"/usr/lib/python3.6/distutils/dist.py\", line 534, in _parse_command_opts\r\n    if not issubclass(cmd_class, Command):\r\nTypeError: issubclass() arg 1 must be a class\r\nTraceback (most recent call last):\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 1041, in <module>\r\n    sys.exit(main())\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 1032, in main\r\n    build_python_wheel(source_dir, build_dir, configs, args.use_cuda, args.use_ngraph, args.use_tensorrt, args.use_openvino, args.use_nuphar, nightly_build)\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 793, in build_python_wheel\r\n    run_subprocess(args, cwd=cwd)\r\n  File \"/tmp/onnxruntime/tools/ci_build/build.py\", line 196, in run_subprocess\r\n    return subprocess.run(args, cwd=cwd, check=True, stdout=stdout, stderr=stderr, env=my_env, shell=shell)\r\n  File \"/usr/lib/python3.6/subprocess.py\", line 438, in run\r\n    output=stdout, stderr=stderr)\r\nsubprocess.CalledProcessError: Command '['/usr/bin/python3', '/tmp/onnxruntime/setup.py', 'bdist_wheel', '--use_cuda']' returned non-zero exit status 1.\r\n```",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/556332771/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/556376161",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-556376161",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 556376161,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NjM3NjE2MQ==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-20T20:49:15Z",
        "updated_at": "2019-11-20T20:49:15Z",
        "author_association": "NONE",
        "body": "On my another computer having Titan V, using the same container got the same problem.\r\n```\r\n[ 98%] Linking CUDA device code CMakeFiles/onnxruntime_pybind11_state.dir/cmake_device_link.o\r\n[100%] Linking CXX shared module onnxruntime_pybind11_state.so\r\n[100%] Built target onnxruntime_pybind11_state\r\n2019-11-20 20:46:57,017 Build [DEBUG] - Running subprocess in '/workspace/onnxruntime/build/Linux/Release'\r\n['/usr/bin/python3', '/workspace/onnxruntime/setup.py', 'bdist_wheel', '--use_cuda']\r\nTraceback (most recent call last):\r\n  File \"/workspace/onnxruntime/setup.py\", line 220, in <module>\r\n    'Programming Language :: Python :: 3.7'],\r\n  File \"/usr/local/lib/python3.6/dist-packages/setuptools/__init__.py\", line 145, in setup\r\n    return distutils.core.setup(**attrs)\r\n  File \"/usr/lib/python3.6/distutils/core.py\", line 134, in setup\r\n    ok = dist.parse_command_line()\r\n  File \"/usr/local/lib/python3.6/dist-packages/setuptools/dist.py\", line 706, in parse_command_line\r\n    result = _Distribution.parse_command_line(self)\r\n  File \"/usr/lib/python3.6/distutils/dist.py\", line 472, in parse_command_line\r\n    args = self._parse_command_opts(parser, args)\r\n  File \"/usr/local/lib/python3.6/dist-packages/setuptools/dist.py\", line 1021, in _parse_command_opts\r\n    nargs = _Distribution._parse_command_opts(self, parser, args)\r\n  File \"/usr/lib/python3.6/distutils/dist.py\", line 534, in _parse_command_opts\r\n    if not issubclass(cmd_class, Command):\r\nTypeError: issubclass() arg 1 must be a class\r\nTraceback (most recent call last):\r\n  File \"/workspace/onnxruntime/tools/ci_build/build.py\", line 1043, in <module>\r\n    sys.exit(main())\r\n  File \"/workspace/onnxruntime/tools/ci_build/build.py\", line 1034, in main\r\n    build_python_wheel(source_dir, build_dir, configs, args.use_cuda, args.use_ngraph, args.use_tensorrt, args.use_openvino, args.use_nuphar, nightly_build)\r\n  File \"/workspace/onnxruntime/tools/ci_build/build.py\", line 795, in build_python_wheel\r\n    run_subprocess(args, cwd=cwd)\r\n  File \"/workspace/onnxruntime/tools/ci_build/build.py\", line 196, in run_subprocess\r\n    return subprocess.run(args, cwd=cwd, check=True, stdout=stdout, stderr=stderr, env=my_env, shell=shell)\r\n  File \"/usr/lib/python3.6/subprocess.py\", line 438, in run\r\n    output=stdout, stderr=stderr)\r\nsubprocess.CalledProcessError: Command '['/usr/bin/python3', '/workspace/onnxruntime/setup.py', 'bdist_wheel', '--use_cuda']' returned non-zero exit status 1.\r\n```",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/556376161/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/556396899",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-556396899",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 556396899,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NjM5Njg5OQ==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-20T21:07:24Z",
        "updated_at": "2019-11-20T21:07:24Z",
        "author_association": "NONE",
        "body": "Removing `--build_wheel` allows to build successfully. What are the potential problems here?\r\n```\r\n./build.sh --config Release --cuda_home /usr/local/cuda --cudnn_home /usr/lib/x86_64-linux-gnu/ --use_cuda --parallel --use_full_protobuf --update --build\r\n```",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/556396899/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/556934572",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-556934572",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 556934572,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NjkzNDU3Mg==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-21T06:04:00Z",
        "updated_at": "2019-11-21T06:04:00Z",
        "author_association": "NONE",
        "body": "This is extremely weird. I uninstalled Python 3.6 in the container, installed Python 3.7 and rebuilt. The error message was gone and the building was successful. While I don't want to use Python 3.7, is there any solution that can keep my Python 3.6? Thanks.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/556934572/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/556934770",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-556934770",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 556934770,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NjkzNDc3MA==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-21T06:04:50Z",
        "updated_at": "2019-11-21T06:04:50Z",
        "author_association": "NONE",
        "body": "I saw in the \"official\" container, you guys are building against Python 3.7.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/556934770/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/558446059",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-558446059",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 558446059,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1ODQ0NjA1OQ==",
        "user": {
            "login": "leimao",
            "id": 17606112,
            "node_id": "MDQ6VXNlcjE3NjA2MTEy",
            "avatar_url": "https://avatars.githubusercontent.com/u/17606112?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/leimao",
            "html_url": "https://github.com/leimao",
            "followers_url": "https://api.github.com/users/leimao/followers",
            "following_url": "https://api.github.com/users/leimao/following{/other_user}",
            "gists_url": "https://api.github.com/users/leimao/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/leimao/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/leimao/subscriptions",
            "organizations_url": "https://api.github.com/users/leimao/orgs",
            "repos_url": "https://api.github.com/users/leimao/repos",
            "events_url": "https://api.github.com/users/leimao/events{/privacy}",
            "received_events_url": "https://api.github.com/users/leimao/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-26T03:43:24Z",
        "updated_at": "2019-11-26T03:43:24Z",
        "author_association": "NONE",
        "body": "Any follow up on this?",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/558446059/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/558557399",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/2436#issuecomment-558557399",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/2436",
        "id": 558557399,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1ODU1NzM5OQ==",
        "user": {
            "login": "skottmckay",
            "id": 979079,
            "node_id": "MDQ6VXNlcjk3OTA3OQ==",
            "avatar_url": "https://avatars.githubusercontent.com/u/979079?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/skottmckay",
            "html_url": "https://github.com/skottmckay",
            "followers_url": "https://api.github.com/users/skottmckay/followers",
            "following_url": "https://api.github.com/users/skottmckay/following{/other_user}",
            "gists_url": "https://api.github.com/users/skottmckay/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/skottmckay/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/skottmckay/subscriptions",
            "organizations_url": "https://api.github.com/users/skottmckay/orgs",
            "repos_url": "https://api.github.com/users/skottmckay/repos",
            "events_url": "https://api.github.com/users/skottmckay/events{/privacy}",
            "received_events_url": "https://api.github.com/users/skottmckay/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-26T10:06:23Z",
        "updated_at": "2019-11-26T10:06:23Z",
        "author_association": "MEMBER",
        "body": "It should work with python 3.6. setup.py however swallows any import errors which might be the cause of your issue. Seems pretty similar to this: https://github.com/intel-isl/Open3D/pull/1012/files\r\n\r\nDo you have the 'wheel' package installed for python 3.6?\r\n\r\nRegarding this: \r\n> It seems that I could still install the Python package even though the test was failed. This misleading messages and tests should really be avoided.\r\n\r\n>> Failed Test Cases:\r\n>> /tmp/onnxruntime/tools/ci_build/build.py:620: UserWarning: onnxmltools and keras are not installed. Following test cannot be run.\r\n>>   warnings.warn(\"onnxmltools and keras are not installed. Following test cannot be run.\")\r\n\r\nThe test isn't actually failing, The first line is actually listing that there were no failed test cases. The following lines are a warning message from the next step in the testing process. I'll see if a message can be inserted in between so it's clearer that the script the 'Failed Test Cases:' is coming from completes. \r\n\r\nFinally, would it be helpful to have a '--skip_tests' parameter that skipped all tests?",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/558557399/reactions",
            "total_count": 1,
            "+1": 1,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    }
]